core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -268,6 +270,31 @@ object UnitAutomation {
         return true
     }
 
+    private fun tryHeadTowardsSiegedCity(unit: MapUnit): Boolean {
+        val siegedCities = unit.civInfo.cities
```

Comment:
.asSequence() first, before the filters

Reply:
Also you can merge the filters to 
`unit.civInfo == it.civInfo && it.health < (it.getMaxHealth() * 0.75) && !it.isInResistance()